### PR TITLE
Refactors item giving

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

ported: https://github.com/tgstation/tgstation/pull/61590

Refactored giving code.
Also adds an option to slam tables.

## Why It's Good For The Game

Bug fix and cool feature

## Testing Photographs and Procedure
I tested it locally, giving worked although I was unable to reproduce the runtime.

## Changelog
:cl: Ryll/Shaps, 
add: You can slam table with *slap emote
fix: fixed out of range bugged message when giving an item
refactor: refactored giving code
/:cl:

### CODE REVIEW

**Comment by ToasterBan** [comment]
we already have high fives, it's just say" `*`highfive and click someone who also did say"`*`highfive

**Comment by PowerfulBacon** [comment]
fill the damn testing bit if you are going to refactor something

**Comment by Archanial** [comment]
ok

File: code/datums/traits/neutral.dm [hunk 1]
```diff
@@ -97,3 +97,80 @@
...
+	return ..()
+
+/// Hook up the specified carbon mob to be offered the item in question, give them the alert and signals and all
+/datum/status_effect/offering/proc/register_candidate(mob/living/carbon/possible_candidate)
+	var/atom/movable/screen/alert/give/G = possible_candidate.throw_alert("[owner]", give_alert_type)
+	if(!G)
+		return
+	LAZYADD(possible_takers, possible_candidate)
+	RegisterSignal(possible_candidate, COMSIG_MOVABLE_MOVED, .proc/check_taker_in_range)
+	G.setup(possible_candidate, owner, offered_item)
```
Comment (PowerfulBacon) [inline code/datums/traits/neutral.dm:149]: Should this register for qdel too to prevent hard deletes?

Comment (Archanial) [reply code/datums/traits/neutral.dm:149]: I think status effects are cleared on mob del but I'd have to double check that

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**